### DISCUSSION TITLE
Sound with DAC is great!

### DISCUSSION THREAD

**Discussion by ChKK1963** [probonopd/MiniDexed#137] [Show and tell]
I received the Raspiaudio+ V2 DAC today - now MiniDexed sounds really great,  no noise or distortion, latency is great as well!

Attached this on top of a GPIO extension hat to be able to easily add the display/encoder once they arrive ...

**Comment by probonopd** [comment]
Does it have a PCM5122 chip? If yes, then it would be great if you could test https://github.com/probonopd/MiniDexed/pull/133 to see whether it breaks your DAC (I hope not). Once you confirm that it is still working, I could merge it.

**Comment by diysynth** [comment]
I guess this is the right place to ask if this DAC will work with minidexed: https://www.berrybase.de/en/raspberry-pi/raspberry-pi-computer/gpio-hats-phats/audio/iqaudio-dac-43

At least it has the PCM5122 DAC that is mandatory.

**Reply by probonopd** [reply]
PCM5122... Then it should probably work.

**Comment by WilliamKWusik** [comment]
What is the latency in MS? Thanks. :-)

**Reply by Banana71** [reply]
Hi William,
The latency of miniDexed depends on many different factors.
- Which RPi model is being used
- The sample rate set in minidexed.ini
- The chunk size set in miniDexed.ini

I can only tell you the differences between a Yamaha SY-99 and a miniDexed. I can't determine the absolute latency without an oscilloscope. For the measurement, I output the note C3 in a DAW. It enters the SY99 via MIDI and is routed to the miniDexed via MIDI True.

The miniDexed ran with the following parameters:
Raspberry Pi 4
minidexed.ini
```
SampleRate=48000
ChunkSize=128
```

I recorded the audio signals from the SY99 on the left and the miniDexed on the right. In the DAW, the SY99 is shown at the top and the miniDexed at the bottom.

**minidexed.ini**
```
SampleRate=48000
ChunkSize=128
```

In this configuration, the miniDexed has a total latency of 11.4 ms and the SY-99 6.7 ms. All we can say for sure is that the miniDexed has a latency that is 4.7ms (11.4 - 6.7) higher than the SY-99.

However, a higher latency also means that even when all other instruments have gone silent, you can still get some sound from the miniDexed. :-) 
All I can say is that I don't notice the latency.

![Latenz SY99 miniDexed](https://github.com/user-attachments/assets/e813797d-2ef1-4b64-97ac-f74c6de52046)

**Reply by WilliamKWusik** [reply]
Thank you so much! :-) And yeah, I figured the 128 chunk size as the latency in samples. 128 samples is actually 2.9 ms, very low. And maybe could go even lower with some optimizations.

**Reply by WilliamKWusik** [reply]
From your test, don't forget to subtract the latency of your audio device. That's the one adding up to near 12 ms latency.

**Reply by WilliamKWusik** [reply]
Ahhh, at 48khz samplerate, 128 samples is 2.6 ms.

**Reply by probonopd** [reply]
Should we change

https://github.com/probonopd/MiniDexed/blob/bba0c93f04f8a40a897faf54378d7f11f2729452/src/config.h#L83

to default to 128?

**Reply by diyelectromusic** [reply]
We shouldn't touch the maximum chunk size in my view.  The default chunk size used is either 1024 or 256: https://github.com/probonopd/MiniDexed/blob/main/src/config.cpp#L66

But the actual chunk size can already be set in miinidexed.ini: https://github.com/probonopd/MiniDexed/blob/main/src/minidexed.ini#L10

But there will be other considerations relating to how much audio is processed per IO scan and the scheduling between cores and stuff like that, so it is better to leave it as something for people to experiment with if their situation and understand allows them to, but leave some usable defaults (as we have right now).

Kevin

**Reply by Banana71** [reply]
> Should we change
> 
> https://github.com/probonopd/MiniDexed/blob/bba0c93f04f8a40a897faf54378d7f11f2729452/src/config.h#L83
> 
> to default to 128?

I don't see any need to limit the maximum chunk size to 128. Or do you want to set the ChuckSize=128 as the default value in the minidexed.ini? I haven't had any problems with a ChuckSize=128 so far, but I've only ever used one type of DAC (GY-PCM5102) and can't comment on other models or the PWM mode.
Perhaps the wiki entries on chunk size and sample rate should be revised.

Does the sample rate still have a direct influence on the reverb time (size), or has this bug been fixed?
